src/libasr/codegen/asr_to_c.cpp

```diff
@@ -490,6 +497,84 @@ R"(
         array_types_decls += open_struct + body + end_struct;
     }
 
+    void visit_EnumType(const ASR::EnumType_t& x) {
+        std::string indent(indentation_level*indentation_spaces, ' ');
+        std::string tab(indentation_spaces, ' ');
+        std::string meta_data = " = {";
+        std::string open_struct = indent + "enum " + std::string(x.m_name) + " {\n";
+        std::string body = "";
+        std::string src_copy = src;
+        int64_t min_value = INT64_MAX;
+        int64_t max_value = INT64_MIN;
+        size_t max_name_len = 0;
+        for( size_t i = 0; i < x.n_members; i++ ) {
+            ASR::symbol_t* member = x.m_symtab->get_symbol(x.m_members[i]);
+            LFORTRAN_ASSERT(ASR::is_a<ASR::Variable_t>(*member));
+            ASR::Variable_t* member_var = ASR::down_cast<ASR::Variable_t>(member);
+            ASR::expr_t* value = ASRUtils::expr_value(member_var->m_symbolic_value);
+            int64_t value_int64 = -1;
+            ASRUtils::extract_value(value, value_int64);
```

Comment:
This will fail if the value is not Integer, correct?

We need to give an error message here. 

I think to do this cleanly, why not store the `enum_type` in ASR directly, and in here, only support `IntegerConsecutiveFrom0` and `IntegerUnique`, and for everything else give an error message "not implemented yet"?

Reply:
> This will fail if the value is not Integer, correct?
> 
> We need to give an error message here.

Yes. Will do.



> I think to do this cleanly, why not store the `enum_type` in ASR directly, and in here, only support `IntegerConsecutiveFrom0` and `IntegerUnique`, and for everything else give an error message "not implemented yet"?

Yes we can but I think its too early to create an `enum_type`. I will wait for all the other cases to be supported and then do this.

Reply:
Done. Added error messages for both non-integer and non-unique integer cases.

Reply:
Perfect.